components/personAvatar.vue

```diff
@@ -1,12 +1,73 @@
 <template>
-  <img v-if="person.avatarThumbnail" :src="person.avatarThumbnail">
-  <img v-else-if="person.avatar" :src="person.avatar">
-  <img v-else-if="person.avatarURL" :src="person.avatarURL">
+  <div class="avatar-holder" v-if="compressedImage" @click="downloadImage">
+    <a :download="person.fullname" ref="imageLink">
+      <img :src="compressedImage">
+      <v-btn class="download-icon" icon>
+        <v-icon class="white--text" >fas fa-arrow-alt-circle-down</v-icon>
+      </v-btn>
+    </a>
+  </div>
   <span v-else :class="person.colorClassName" class="white--text headline">{{person.firstLetter}}</span>
 </template>
 
 <script>
   export default {
-    props: ['person']
+    props: ['person'],
+    data () {
+      const { avatar, avatarURL } = this.person
+      return {
+        image: avatar || avatarURL
+      }
+    },
+    watch: {
```

Comment:
@ssoonmi what's the purpose of the watch here?

Reply:
the person changes in the component when you open a contact, then open another person's contact. `this.image` needs to be updated when person changes.